src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,49 +1,105 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_ARRAY_CAPACITY = 10;
+    private T[] array;
+    private int sizeOfFilledArray;
+
+    public ArrayList() {
+        this.array = (T[]) new Object[DEFAULT_ARRAY_CAPACITY];
+        this.sizeOfFilledArray = 0;
+    }
+
+    public void growCapacity() {
+        T[] newArray = (T[]) new Object[sizeOfFilledArray + DEFAULT_ARRAY_CAPACITY / 2];
```

Comment:
this is hardcode - we should take the actual `array` length and increase it 1.5 times

DEFAULT_ARRAY_CAPACITY / 2 will always be 5, no matter which is the size of your array